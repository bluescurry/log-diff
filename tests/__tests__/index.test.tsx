import { render } from 'enzyme';
import React from 'react';
import { List } from 'antd';
import LogDiff from '../../src/index';

describe('BasicLogDiff', () => {
  it('🎏 base use', () => {
    const initalData = {
      before: {
        basic: {
          name: '众包挂靠',
          url: 'http://erp-1258916733.cos.ap-shanghai.myqcloud.com/manage/temp/378bbd5adad4e1f0266518522c559e42.mov',
          is_long: '0',
        }
      },
      after: {
        basic: {
          is_long: '1',
        }
      },
      constants: {
        is_long: {
          '0': '否',
          '1': '是',
        }
      },
    }
    const html = render(
      <LogDiff
        data={initalData}
      >
        <LogDiff.Input label="合同名称" name={['basic', 'name']} />
        <LogDiff.Image label="身份证人像面照片" name={['basic', 'url']} />
        <LogDiff.Radio label="有效期限是否长期" name={['basic', 'is_long']} />
      </LogDiff>,
    );
    expect(html).toMatchSnapshot();
  });

  it('🎏 Combination use ', () => {
    const initalData = {
      before: {
        basic: {
          start: '2020-05-25',
          end: '2020-05-27',
        }
      },
      after: {
        basic: {
          start: '2020-05-28',
          end: '2020-05-29',
        }
      },
      constants: {
      },
    }

    const html = render(
      <LogDiff
        data={initalData}
      >
        <LogDiff.Input label="期间" name={[['basic', 'start'], ['basic', 'end']]} />
      </LogDiff>,
    );
    expect(html).toMatchSnapshot();
  });

  it('🎏 list use ', () => {
    const initalData = {
      before: {
        business: {
          details: [
            {
              area: [ 'tc311y', 'tc755y' ],
              license: 'a287'
            },
            {
              area: [
                'tc755y'
              ],
              license: 'a273'
            }
          ],
        },
      },
      after: {
        business: {
          details: [
          ],
        },
      },
      constants: {
        area: {
          'tc311y': '北京',
          'tc755y': '上海',
        }
      },
    }
    
    const businessDetailData = [0, 1];

    const html = render(
      <LogDiff
        data={initalData}
      >
        <List
          size="small"
          bordered
          dataSource={businessDetailData}
          renderItem={item => (
            <List.Item
            >
              <LogDiff.Radio label="地区" name={['business', 'details', item, 'area']} />
              <LogDiff.Input label="证书" name={['business', 'details', item, 'license']} />
            </List.Item>
          )}
        />
      </LogDiff>,
    );
    expect(html).toMatchSnapshot();
  });
});
