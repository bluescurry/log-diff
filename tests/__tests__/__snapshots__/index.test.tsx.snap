// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLogDiff 🎏 Combination use  1`] = `
<div
  class="space-divide"
>
  <span
    class="ant-typography log-diff-item-label"
  >
    <span
      class="label-sign"
    >
      期间
    </span>
  </span>
  <span
    class="ant-typography"
  >
    <span
      class="pass space"
    >
      <span
        class="space"
      >
        2020-05-25
      </span>
      <span
        class="space"
      >
        2020-05-27
      </span>
    </span>
    <span
      class="now"
    >
      <span
        class="space"
      >
        2020-05-28
      </span>
      <span
        class="space"
      >
        2020-05-29
      </span>
    </span>
  </span>
</div>
`;

exports[`BasicLogDiff 🎏 base use 1`] = `
Array [
  <div
    class="space-divide"
  >
    <span
      class="ant-typography log-diff-item-label"
    >
      <span
        class="label-sign"
      >
        合同名称
      </span>
    </span>
    <span
      class="ant-typography"
    >
      <span
        class="pass space"
      >
        <span
          class="space"
        >
          众包挂靠
        </span>
      </span>
      <span
        class="now"
      >
        -
      </span>
    </span>
  </div>,
  <div
    class="space-divide"
  >
    <span
      class="ant-typography log-diff-item-label"
    >
      <span
        class="label-sign"
      >
        身份证人像面照片
      </span>
    </span>
    <span
      class="ant-typography"
    >
      <span
        class="pass space"
      >
        <a
          href="http://erp-1258916733.cos.ap-shanghai.myqcloud.com/manage/temp/378bbd5adad4e1f0266518522c559e42.mov"
        >
          http://erp-1258916733.cos.ap-shanghai.myqcloud.com/manage/temp/378bbd5adad4e1f0266518522c559e42.mov
        </a>
      </span>
      <span
        class="now"
      >
        -
      </span>
    </span>
  </div>,
  <div
    class="space-divide"
  >
    <span
      class="ant-typography log-diff-item-label"
    >
      <span
        class="label-sign"
      >
        有效期限是否长期
      </span>
    </span>
    <span
      class="ant-typography"
    >
      <span
        class="pass space"
      >
        <span
          class="space"
        >
          否
        </span>
      </span>
      <span
        class="now"
      >
        <span
          class="space"
        >
          是
        </span>
      </span>
    </span>
  </div>,
]
`;

exports[`BasicLogDiff 🎏 list use  1`] = `
<div
  class="ant-list ant-list-sm ant-list-split ant-list-bordered"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="space-divide"
          >
            <span
              class="ant-typography log-diff-item-label"
            >
              <span
                class="label-sign"
              >
                地区
              </span>
            </span>
            <span
              class="ant-typography"
            >
              <span
                class="pass space"
              >
                <span
                  class="space"
                >
                  北京
                </span>
                <span
                  class="space"
                >
                  上海
                </span>
              </span>
              <span
                class="now"
              >
                -
              </span>
            </span>
          </div>
          <div
            class="space-divide"
          >
            <span
              class="ant-typography log-diff-item-label"
            >
              <span
                class="label-sign"
              >
                证书
              </span>
            </span>
            <span
              class="ant-typography"
            >
              <span
                class="pass space"
              >
                <span
                  class="space"
                >
                  a287
                </span>
              </span>
              <span
                class="now"
              >
                -
              </span>
            </span>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="space-divide"
          >
            <span
              class="ant-typography log-diff-item-label"
            >
              <span
                class="label-sign"
              >
                地区
              </span>
            </span>
            <span
              class="ant-typography"
            >
              <span
                class="pass space"
              >
                <span
                  class="space"
                >
                  上海
                </span>
              </span>
              <span
                class="now"
              >
                -
              </span>
            </span>
          </div>
          <div
            class="space-divide"
          >
            <span
              class="ant-typography log-diff-item-label"
            >
              <span
                class="label-sign"
              >
                证书
              </span>
            </span>
            <span
              class="ant-typography"
            >
              <span
                class="pass space"
              >
                <span
                  class="space"
                >
                  a273
                </span>
              </span>
              <span
                class="now"
              >
                -
              </span>
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
